library/src/main/java/com/paymentkit/views/FieldHolder.java

```diff
@@ -129,108 +88,24 @@ private void setNecessaryFields() {
         setAddStatesFromChildren(true);
     }
 
-    private void setExtraFieldsAlpha() {
-        ViewHelper.setAlpha(mExpirationEditText, 0.0f);
-        mExpirationEditText.setVisibility(View.GONE);
-    }
-
-    private void setCardEntryListeners() {
-        mExpirationEditText.setCardEntryListener(mCardEntryListener);
-    }
-
     private void validateCard() {
         String stringNumber = getCardNumber();
         long cardNumber = !TextUtils.isEmpty(stringNumber) ? Long.parseLong(stringNumber) : 0;
         if (ValidateCreditCard.isValid(cardNumber)) {
-            CardType cardType = ValidateCreditCard.matchCardType(cardNumber);
-            mCardIcon.setCardType(cardType);
-            transitionToExtraFields();
-        } else {
-            mCardHolder.indicateInvalidCardNum();
-        }
-    }
+            mCardHolder.setDrawable(ValidateCreditCard.matchCardType(cardNumber));
 
```

Comment:
Is this just removing code?

Reply:
yes. We are not using the cardIcon class anymore and there are no fields to transition to now when the card is valid.